File: scripts/gps_publisher.py
#!/usr/bin/env python

import rospy
from first_project.msg import GPS_data
import random

#create a new publisher. we specify the topic name, then type of message then the queue size
pub = rospy.Publisher('gps_topic', GPS_data, queue_size=10)

#we need to initialize the node
rospy.init_node('gps_publisher_node', anonymous=True)

#set the loop rate
rate = rospy.Rate(1) # 1hz
#keep publishing until a Ctrl-C is pressed
i = 0
while not rospy.is_shutdown():
    GPS = GPS_data()
    GPS.longitude = 24.33 + (random.random()*2)
    GPS.latitude = 33.41 + (random.random()*2)
    GPS.heading = (random.random()*2)
    rospy.loginfo("I publish:")
    rospy.loginfo(GPS)
    pub.publish(GPS)
    rate.sleep()
    i=i+1

File: src/gps_publisher.cpp
#include "ros/ros.h"
#include "first_project/GPS_data.h"

int main (int argc, char **argv) {
    // Initiate a new ROS node named "gps_publisher_node"
	ros::init(argc, argv, "gps_publisher_node");
    //create a node handle: it is reference assigned to a new node
	ros::NodeHandle nh;
    //Rate is a class the is used to define frequency for a loop. Here we send a message each two seconds.
	ros::Rate loop_rate(10);
    //create a publisher with a topic "gps_topic" that will send a String message
	ros::Publisher pub = nh.advertise<first_project::GPS_data>("gps_topic", 1000);

    ROS_INFO("Test node has been started");

    int i = 0;
    while(ros::ok()){
        first_project::GPS_data gps;
        gps.longitude = 24.33 + (double)(i);
        gps.latitude = 33.41 + (double)(i);
        gps.heading = 3.51 + (double)(i);

        ROS_INFO("I publish longitude: %.6f", gps.longitude);
        ROS_INFO("I publish latitude: %.6f", gps.latitude);
        ROS_INFO("I publish bearing: %.6f", gps.heading);

        pub.publish(gps);
        loop_rate.sleep();
        i++;
    }
}